src/Neo4jClient.Extension/Cypher/Extension/CypherExtension.Main.cs

```diff
@@ -75,13 +75,44 @@ public static ICypherFluentQuery OptionalMatchEntity<T>(this ICypherFluentQuery
             return query;
         }
 
-        public static ICypherFluentQuery MergeEntity<T>(this ICypherFluentQuery query, T entity, string paramKey = null, List<CypherProperty> mergeOverride = null, List<CypherProperty> onMatchOverride = null, List<CypherProperty> onCreateOverride = null,string preCql = "", string postCql = "") where T : class
+        public static ICypherFluentQuery MergeEntity<T>(this ICypherFluentQuery query, T entity, string identifier = null, List<CypherProperty> mergeOverride = null, List<CypherProperty> onMatchOverride = null, List<CypherProperty> onCreateOverride = null,string preCql = "", string postCql = "") where T : class
+        {
+            var options = new MergeOptions
+            {
+                Identifier = entity.EntityParamKey(identifier),
+                PreCql = preCql,
+                PostCql = postCql,
+                MergeOverride = mergeOverride,
+                OnCreateOverride = onCreateOverride,
+                OnMatchOverride = onMatchOverride
+            };
+            return MergeEntity(query, entity, options);
+        }
+
+        public static ICypherFluentQuery MergeEntity<T>(this ICypherFluentQuery query, T entity, MergeOptions options) where T : class
         {
-            paramKey = entity.EntityParamKey(paramKey);
             var context = CypherExtensionContext.Create(query);
-            var cypher1= entity.ToCypherString<T, CypherMergeAttribute>(context, paramKey, mergeOverride);
-            var cql = string.Format("{0}({1}){2}", preCql, cypher1, postCql);
-            return query.CommonMerge(entity, paramKey, cql, mergeOverride, onMatchOverride, onCreateOverride);
+            string pattern;
+            string wrappedPattern;
+
+            if (options.MergeViaRelationship != null)
```

Comment:
only the setting of `pattern` is affected by this `if` statement `wrappedPattern = string.Format("{0}({1}){2}", options.PreCql, pattern, options.PostCql);` is common